packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts

```diff
@@ -1299,47 +1291,57 @@ export class AuthTransformer extends TransformerAuthBase implements TransformerA
   };
 
   addAutoGeneratedHasManyFields = (
+    ctx: TransformerContextProvider,
     typeDefinitions: ObjectTypeDefinitionNode[],
     def: ObjectTypeDefinitionNode,
     allowedFields: Set<string>,
   ): void => {
     const hasManyRelatedFields = typeDefinitions
-      .map(it => it.fields.map(field => ({ ...field, relatedType: it.name.value })))
+      .map(it => it.fields.map(field => ({ ...field, relatedType: it })))
       .flat()
       .filter(it => getBaseType(it.type) === def.name.value && it.directives?.some(d => d.name.value === 'hasMany'));
 
     hasManyRelatedFields.forEach(relatedField => {
-      allowedFields.add(getConnectionAttributeName(relatedField.relatedType, relatedField.name.value));
+      allowedFields.add(
+        getConnectionAttributeName(
+          ctx.featureFlags,
+          relatedField.relatedType.name.value,
+          relatedField.name.value,
+          getObjectPrimaryKey(relatedField.relatedType).name.value,
+        ),
+      );
+      getSortKeyFieldNames(relatedField.relatedType).forEach(sortKeyFieldName => {
+        allowedFields.add(
+          getSortKeyConnectionAttributeName(
+            ctx.featureFlags,
+            relatedField.relatedType.name.value,
+            relatedField.name.value,
+            sortKeyFieldName,
+          ),
+        );
+      });
     });
   };
 
   /**
    * Helper to add auto generated @hasOne fields
    */
-  static addAutoGeneratedHasOneFields(
+  addAutoGeneratedHasOneFields = (
+    ctx: TransformerContextProvider,
     typeDefinitions: ObjectTypeDefinitionNode[],
     fields: readonly string[],
     def: ObjectTypeDefinitionNode,
     allowedFields: Set<string>,
-  ): void {
+  ): void => {
     fields.forEach(field => {
       const modelField = def.fields.find(it => it.name.value === field);
 
       const directives = modelField.directives?.filter(
         dir => !dir.arguments?.some(it => it.name.value === 'fields') && (dir.name.value === 'hasOne' || dir.name.value === 'belongsTo'),
       );
-      directives.forEach(directive => {
-        if (directive.name.value === 'hasOne') {
-          allowedFields.add(getConnectionAttributeName(def.name.value, field));
-        } else if (directive.name.value === 'belongsTo') {
-          const relatedType = typeDefinitions.find(
-            it => it.name.value === getBaseType(modelField.type)
-              && it.fields?.some(f => getBaseType(f.type) === def.name.value && f.directives?.some(d => d.name.value === 'hasOne')),
-          );
-          if (relatedType) {
-            allowedFields.add(getConnectionAttributeName(def.name.value, field));
-          }
-        }
+      directives.forEach(_ => {
```

Comment:
Can I just get some context on this area? why were we doing directive specific checks before and why do we not need them now?

Reply:
this was a miss on my end, updated and added a test, see last commit